### PR TITLE
Setup React Redux Flow

### PR DESCRIPTION
**Pull request by itsgratien**
#### What does this PR do?
This will setup react redux boilerplate in our project (Authors Haven--frontend).

#### Description of Task to be completed?

- Installed packages: redux, redux-devtools-extension, react-redux, redux-thunk, prop-types
- setup boilerplate of using react-redux
- created redux store
- Tested this chore and made sure that everything is working

#### How should this be manually tested?

- After cloning the repository with [bad_ass-ah-frontend](https://github.com/andela/bad_ass-ah-frontend), checkout to this branch `ch-164489879-setupReactReduxFlow`
- Run dependencies installation with `npm install`
- start the server with `npm start`
#### Screenshots (if appropriate)
Screen Shot 2019-05-14 at 13 01 19

##### Creating Store
Screen Shot 2019-05-14 at 13 06 23

##### Dumy Data from actions to reducer
Screen Shot 2019-05-14 at 13 02 19

#### What are the relevant pivotal tracker stories?
[164489879](https://www.pivotaltracker.com/story/show/164489879)

### CODE REVIEW

File: src/components/article/AllArticles.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React ,{useEffect,Fragment} from 'react'
+import propTypes from 'prop-types';
+import {connect} from "react-redux";
+import {getAllArticle} from "../../actions/article";
+
+
+function AllArticles({getAllArticle, article:{articles}}) {
```
Comment (nakiwuge) [inline src/components/article/AllArticles.js]: use Es6 for the function

File: src/components/article/AllArticles.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React ,{useEffect,Fragment} from 'react'
+import propTypes from 'prop-types';
+import {connect} from "react-redux";
+import {getAllArticle} from "../../actions/article";
+
```
Comment (nakiwuge) [inline src/components/article/AllArticles.js]: remove this line

File: src/reducers/article.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//@call type going to be used..
+import {GET_ALL_ARTICLE} from "../actions/index";
+const initialState=[];
+
+export function getAllArticle(state=initialState, action){
```
Comment (nakiwuge) [inline src/reducers/article.js]: use ES6

**Comment by Jaman-dedy** [comment]
You've achieved a great Job @itsgracian , however for the best practice, I can suggest you to create an  other file in actions where we can define all our constants to be used in all files; as the reducers and actions will use the same variables like `TYPE` it can be good to define all our constant in one file.. thank

File: src/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+//@here we will combine all reducers file 
+import {combineReducers} from "redux";
+import {getAllArticle} from "./article";
+
+//@create root reducer
+
```
Comment (nakiwuge) [inline src/reducers/index.js:6]: remove empty line

File: src/components/article/AllArticles.js (hunk 1 above)
Comment (itsgratien) [reply src/components/article/AllArticles.js]: Thanks, @zellymiriam, I changed it

**Comment by itsgratien** [comment]
thanks, @Jaman-dedy but I think all action type is in one file, check it again carefully.

File: src/components/articles/AllArticles.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React ,{useEffect,Fragment} from 'react'
+import propTypes from 'prop-types';
+import {connect} from "react-redux";
+import {getAllArticle} from "../../actions/article";
+import Navbar from "../layouts/navbar";
+const AllArticles=({getAllArticle, article:{articles}})=> {
```
Comment (nakiwuge) [inline src/components/articles/AllArticles.js:7]: add a new line before this line

File: src/actions/article.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+//@call type we are going to use
+import {GET_ALL_ARTICLE} from "./type";
+const articles=[{
```
Comment (nakiwuge) [inline src/actions/article.js:4]: Add a new line before this line

File: src/Store.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+//@redux store
+import {createStore, applyMiddleware} from "redux";
+import {composeWithDevTools} from "redux-devtools-extension";
+import thunk from "redux-thunk";
+import rootReducers from "./reducers/index";
+//@middleware
```
Comment (nakiwuge) [inline src/Store.js:7]: Add a new line before this line

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**